File: Backend/Model/Room.go
package model

import (
	"github.com/gorilla/websocket"
)

type Room struct {
	Id                int
	PlayerConnections map[string]*websocket.Conn
	Messages          [][]string // of the form [[username: message], [username: message], [username, message]] in order
	Owner             string     // the playerID of the user who created the room
	Password          string     // "" if no password, otherwise the password to a room for a private room
}

File: frontend/src/classes/Room.ts
import { EventEmitter } from "events";

export default class Room extends EventEmitter {
  public id: number;
  public playersInRoom: string[]
  public messages: string[][]
  public owner: string
  public password: string

  constructor(
    id: number,
    playersInRoom?: string[],
    messages?: string[][],
    owner?: string,
    password?: string
  ) {
    super();
    this.id = id;
    this.playersInRoom = playersInRoom ? playersInRoom : []
    this.messages = messages ? messages : []
    this.owner = owner ? owner : ""
    this.password = password ? password : ""
  }
}
